File: common/src/main/java/org/valkyrienskies/clockwork/content/contraptions/phys/bearing/PhysBearingCreateData.java
package org.valkyrienskies.clockwork.content.contraptions.phys.bearing;

import org.joml.Vector3d;
import org.joml.Vector3dc;
import org.valkyrienskies.core.api.ships.Ship;
import org.valkyrienskies.core.apigame.constraints.VSConstraintAndId;

public record PhysBearingCreateData(Vector3dc bearingPos, Vector3dc bearingAxis, double bearingAngle, float bearingRPM, boolean locked, long shiptraptionID, VSConstraintAndId constraint, VSConstraintAndId hingeConstraint, VSConstraintAndId posDampConstraint, VSConstraintAndId rotDampConstraint) {
}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/content/contraptions/phys/bearing/data/PhysBearingCreateData.kt
package org.valkyrienskies.clockwork.content.contraptions.phys.bearing.data

import org.joml.Vector3dc
import org.valkyrienskies.core.apigame.joints.VSJointAndId

data class PhysBearingCreateData(
    val bearingPos: Vector3dc, // TODO useless
    val bearingAxis: Vector3dc,
    val bearingAngle: Double,
    val bearingRPM: Float,
    val locked: Boolean,
    val shiptraptionID: Long,
    val constraint: VSJointAndId,
)
